lib/reversetunnel/agentpool.go

```diff
@@ -675,7 +675,8 @@ func (c *agentPoolRuntimeConfig) updateRemote(ctx context.Context, addr *utils.N
 		if ok := errors.As(err, &tls.RecordHeaderError{}); !ok {
 			return trace.Wrap(err)
 		}
-	} else {
+		// Only use the ping results if they weren't from a minimal api handler on the reverse tunnel port.
+	} else if addr.Addr != ping.Proxy.SSH.TunnelListenAddr || ping.Proxy.SSH.TunnelListenAddr == ping.Proxy.SSH.WebListenAddr {
```

Comment:
I'm finding this hard to follow.

1. What is it about the minimal API handler that makes us unable to do TLS routing?
2. Could we get some unambiguous data from Ping that makes this situation easier to detect?

Reply:
The minimal API handler only exists when the proxy and reverse tunnel are listening on separate ports, so it will never do TLS routing.

Reply:
Let's simplify and incorporate your response as a comment. See my suggested change below.